ipalib/x509.py

```diff
@@ -488,6 +489,77 @@ def load_private_key_list(data, password=None):
     return priv_keys
 
 
+def load_certificates_from_file(filename):
+    """
+    Load a certificate list from a file. The file may be a PEM or
+    DER certificate or a PKCS#7 certificate chain.
+
+    Return a list of python-cryptography ``Certificate`` objects.
+    """
+    extracted_certs = []
+
+    try:
+        with open(filename, 'rb') as f:
+            data = f.read()
+    except IOError as e:
+        raise RuntimeError(
+            "Failed to open %s: %s" % (filename, e.strerror))
+
+    # Try to parse the file as PEM file
+    matches = list(
+        re.finditer(
+            br'-----BEGIN (.+?)-----(.*?)-----END \1-----',
+            data, re.DOTALL
+        )
+    )
+    if matches:
+        for match in matches:
+            body = match.group()
+            label = match.group(1)
+            line = len(data[:match.start() + 1].splitlines())
+
+            if label in (b'CERTIFICATE', b'X509 CERTIFICATE',
```

Comment:
There is also ``TRUSTED CERTIFICATE``, which is a certificate with additional trust information. Fedora uses it in ``ca-bundle.trust.crt``

Reply:
Adding this is an RFE. I'm just moving code around, not extending features.